src/FakeItEasy.Analyzer/SymbolExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.CodeAnalysis;
+
+    public static class SymbolExtensions
+    {
+        public static string GetFullName(this INamedTypeSymbol type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            var stack = new Stack<string>();
+            stack.Push(type.Name);
+            var tmp = type.ContainingType;
+            while (tmp != null)
+            {
+                type = tmp;
+                stack.Push(tmp.Name);
+                tmp = tmp.ContainingType;
+            }
+
+            var ns = type.ContainingNamespace;
+            while (!string.IsNullOrEmpty(ns?.Name))
+            {
+                stack.Push(ns.Name);
+                ns = ns.ContainingNamespace;
+            }
+
+            return string.Join(".", stack);
+        }
+    }
+}

```

Comment:
@thomaslevesque, now I feel bad. (And at the risk of compounding things, I liked the original version of your comment better. I thought you were very funny!)

The code is certainly understandable, and very terse names work well enough given the limited scope of the method. It's just that my inclination would've been to choose names that were a little more descriptive. (Which is in the eye of the beholder, of course. It's all style.)

As an example of the sort of name I might've picked, and I'm not saying "Take these! Take these!", I'd probably go with something more like

```c#
var nameParts = new Stack<string>();
nameParts.Push(type.Name);
var containingType = type.ContainingType;
while (containingType != null)
{
    nameParts.Push(containingType.Name);
    type = containingType;
    containingType = containingType.ContainingType;
}

var typeNamespace = type.ContainingNamespace;
while (!string.IsNullOrEmpty(typeNamespace?.Name))
{
    nameParts.Push(typeNamespace.Name);
    typeNamespace = typeNamespace.ContainingNamespace;
}

return string.Join(".", nameParts);
```

although I don't love `typeNamespace`. I'd had `@namespace`, but I try to avoid `@`s. They feel dirty to me.

Maybe this is no better.

As an aside, I also moved the assignment `type = containingType` down one line because it felt better to me to have the 2 "walking to the outside of the code" lines next to each other.

I'd wondered whether you even needed to keep track of `type` inside the first loop, since I would have guessed that the namespace would be the same. I tried to experiment with a call of the form `OuterClass.InnerClass.Method()`, but was not smart enough to get to the point where `GetFullName` would be called. :confounded: